azurerm/internal/services/appservice/helpers/shared_schema.go

```diff
@@ -0,0 +1,1167 @@
+				},
+
+				"runtime_version": {
+					Type:     pluginsdk.TypeString,
+					Optional: true,
+					Computed: true,
+				},
+
+				"token_refresh_extension_hours": {
+					Type:     pluginsdk.TypeFloat,
+					Optional: true,
+					Default:  72,
+				},
+
+				"token_store_enabled": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					Default:  false,
+				},
+
+				"unauthenticated_client_action": {
+					Type:     pluginsdk.TypeString,
+					Optional: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						string(web.UnauthenticatedClientActionAllowAnonymous),
+						string(web.UnauthenticatedClientActionRedirectToLoginPage),
+					}, false),
+				},
+
+				"active_directory": AadAuthSettingsSchema(),
+
+				"facebook": FacebookAuthSettingsSchema(),
+
+				"github": GithubAuthSettingsSchema(),
+
+				"google": GoogleAuthSettingsSchema(),
+
+				"microsoft": MicrosoftAuthSettingsSchema(),
+
+				"twitter": TwitterAuthSettingsSchema(),
+			},
+		},
+	}
+}
+
+type AadAuthSettings struct {
+	ClientId                string   `tfschema:"client_id"`
+	ClientSecret            string   `tfschema:"client_secret"`
+	ClientSecretSettingName string   `tfschema:"client_secret_setting_name"`
+	AllowedAudiences        []string `tfschema:"allowed_audiences"`
+}
+
+func AadAuthSettingsSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"client_id": {
+					Type:     pluginsdk.TypeString,
+					Required: true,
+				},
+
+				"client_secret": {
+					Type:      pluginsdk.TypeString,
+					Optional:  true,
+					Sensitive: true,
+					ExactlyOneOf: []string{
+						"auth_settings.0.active_directory.0.client_secret",
+						"auth_settings.0.active_directory.0.client_secret_setting_name",
+					},
+				},
+
+				"client_secret_setting_name": {
+					Type:     pluginsdk.TypeString,
+					Optional: true,
+					ExactlyOneOf: []string{
+						"auth_settings.0.active_directory.0.client_secret",
+						"auth_settings.0.active_directory.0.client_secret_setting_name",
+					},
+				},
+
+				"allowed_audiences": {
+					Type:     pluginsdk.TypeList,
+					Optional: true,
+					Elem: &pluginsdk.Schema{
+						Type: pluginsdk.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+type FacebookAuthSettings struct {
+	AppId                string   `tfschema:"app_id"`
+	AppSecret            string   `tfschema:"app_secret"`
+	AppSecretSettingName string   `tfschema:"app_secret_setting_name"`
+	OauthScopes          []string `tfschema:"oauth_scopes"`
+}
+
+func FacebookAuthSettingsSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*pluginsdk.Schema{
+				"app_id": {
+					Type:     pluginsdk.TypeString,
+					Required: true,
+				},
```

Comment:
should we validate this?

Reply:
This is the Facebook App Id, not an Azure/Resource ID, so _probably_ not?